sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -2242,6 +2242,29 @@ abstract class SQLQuerySuiteBase extends QueryTest with SQLTestUtils with TestHi
     }
   }
 
+  test("SPARK-32889 ORC table column name supports special characters like $ eg.") {
```

Comment:
Please move this test into `FileBasedDataSourceSuite.scala` and generalize it for all data sources like the following.
```scala
allFileBasedDataSources.foreach { format =>
```

Reply:
thank you. @dongjoon-hyun 
In data source. orc and json column name almost supports all characters. 
parquet support all except `.*[ ,;{}()\n\t=].*`
csv doesn't support `\r \n` and so on.
they are all different.  this pr is make orc sql support special chars. shoud we add orc unit test in FileBasedDataSourceSuite only in this pr ?

Reply:
Yep. You can add a single test case for all supporting formats.